components/common/src/util/path.rs

```diff
@@ -19,9 +19,13 @@ use std::io::BufReader;
 use std::path::PathBuf;
 use std::str::FromStr;
 
+use crate::command::package::install::{self, InstallHookMode, InstallMode, LocalPackageUsage};
 use crate::error::{Error, Result};
-use crate::hcore::fs::{find_command, FS_ROOT_PATH};
-use crate::hcore::package::{PackageIdent, PackageInstall};
+use crate::hcore::fs::{cache_artifact_path, find_command, FS_ROOT_PATH};
+use crate::hcore::package::{PackageIdent, PackageInstall, PackageTarget};
+use crate::hcore::url::default_bldr_url;
+use crate::hcore::{channel, PROGRAM_NAME};
```

Comment:
One pattern that I've found useful lately is to write these imports like this:

```rust
use habitat_core::{
    fs::{cache_artifact_path, find_command, FS_ROOT_PATH},
    package::{PackageIdent, PackageInstall, PackageTarget},
    url::default_bldr_url,
    channel, PROGRAM_NAME,
};
```
I find the `crate::hcore` bit a little misleading, because `hcore` is really just an alias we have in the root of this crate to point to `habitat_core`. With the changes in Rust 2018, we can change things up to what I've got above.

The nice thing about this is that you can clearly distinguish between what's from the current crate (it starts with `crate::`) and what isn't.

Similarly, you can group everything from this crate in a `use crate::{ ... };`  statement. And if you put all the `use` statements together with no intervening newlines, `rustfmt` will keep everything nicely organized in a very sensible way.